### PR TITLE
Add binary op gcd

### PR DESCRIPTION
**Pull request by zasdfgbnm**
The main purpose for adding this op is to use it to fix https://github.com/NVIDIA/Fuser/issues/413, but since this op is supported by PyTorch, so why not expose it to the user?

### CODE REVIEW

**Comment by zasdfgbnm** [comment]
!build

File: csrc/dynamic_type.h
Comment (zasdfgbnm) [inline csrc/dynamic_type.h:1]: The evaluator path is not tested in this PR, but I will use it soon in my PR that fixes https://github.com/NVIDIA/Fuser/issues/413

**Comment by zasdfgbnm** [comment]
!build

File: .gitignore [hunk 1]
```diff
@@ -42,3 +42,4 @@ nvfuser/cmake
 
 # debugging temporaries
 *.log
+foo.bin
```
Comment (naoyam) [inline .gitignore:45]: Was this just added accidentally?

Comment (zasdfgbnm) [reply .gitignore:45]: This is not. Some test generate this, I am not sure, but probably related to serialization or kernel db.

**Pull request merged**